Sources/Kitura-CredentialsTwitter/CredentialsTwitter.swift

```diff
@@ -0,0 +1,358 @@
+
+    private var oAuthToken: String = ""
+    private var oAuthTokenSecret: String = ""
+    private var oAuthTokenVerifier: String = ""
+
+    
+    /// Initialization method used to gather the consumer key and consumer secret used by Twitter.
+    ///
+    /// - Parameters:
+    ///   - consumerKey: The consumer key gathered from https://apps.twitter.com for your application.
+    ///   - consumerSecret: The consumer secret gathered from https://apps.twitter.com for your application.
+    public init(consumerKey: String,
+                consumerSecret: String) {
+        self.consumerKey = consumerKey
+        self.consumerSecret = consumerSecret
+    }
+
+    /// CredentialsPluginProtocol
+
+    /// The name of the plugin.
+    public var name: String {
+        return "Twitter"
+    }
+
+    /// User profile cache.
+    public var usersCache: NSCache<NSString, BaseCacheElement>?
+
+    /// An indication as to whether the plugin is redirecting or not.
+    public var redirecting: Bool {
+        return true
+    }
+
+    /// A delegate for `UserProfile` manipulation.
+    public var userProfileDelegate: UserProfileDelegate?
+
+    /// Authenticate an incoming request.
+    ///
+    /// - Parameter request: The `RouterRequest` object used to get information
+    ///                     about the request.
+    /// - Parameter response: The `RouterResponse` object used to respond to the
+    ///                       request.
+    /// - Parameter options: The dictionary of plugin specific options.
+    /// - Parameter onSuccess: The closure to invoke in the case of successful authentication.
+    /// - Parameter onFailure: The closure to invoke in the case of an authentication failure.
+    /// - Parameter onPass: The closure to invoke when the plugin doesn't recognize the
+    ///                     authentication data (usually an authentication token) in the request.
+    /// - Parameter inProgress: The closure to invoke to cause a redirect to the login page in the
+    ///                     case of redirecting authentication.
+    public func authenticate (request: RouterRequest,
+                              response: RouterResponse,
+                              options: [String:Any],
+                              onSuccess: @escaping (UserProfile) -> Void,
+                              onFailure: @escaping (HTTPStatusCode?, [String:String]?) -> Void,
+                              onPass: @escaping (HTTPStatusCode?, [String:String]?) -> Void,
+                              inProgress: @escaping () -> Void) {
+        // If Step 2, is accomplished, the query will have `oauth_verifier` and `oauth_token`. The oauth_token should
+        // match the OAuthToken you got during Step 1.
+        if let verifier = request.queryParameters["oauth_verifier"],
+            let newOAuthToken = request.queryParameters["oauth_token"],
+            newOAuthToken == oAuthToken {
+            oAuthTokenVerifier = verifier
```

Comment:
I've found I have to override Xcode (and manually indent) for multiline if-lets but not for multiline guard-lets